src/tests/unit_tls.cpp

```diff
@@ -444,183 +501,234 @@ Test::Result test_dtls_handshake(Botan::TLS::Protocol_Version offer_version,
+                                         next_protocol_chooser,
+                                         true));
+
+               // TLS::Client object constructed by legacy constructor.
+               client.reset(
+                  new Botan::TLS::Client(queue_inserter(c2s_traffic),
+                                         queue_inserter(client_recv),
+                                         alert_cb_with_data, 
+                                         handshake_complete,
+                                         client_sessions,
+                                         creds,
+                                         policy,
+                                         rng,
+                                         Botan::TLS::Server_Information("server.example.com"),
+                                         offer_version,
+                                         protocols_offered));
                }
 
-            if(handshake_done && (client.is_closed() || server.is_closed()))
-               break;
-
-            if(client.is_active() && client_sent.empty())
+            while(true)
                {
-               // Choose a len between 1 and 511 and send random chunks:
-               const size_t c_len = 1 + rng.next_byte() + rng.next_byte();
-               client_sent = unlock(rng.random_vec(c_len));
+               // TODO: client and server should be in different threads
+               std::this_thread::sleep_for(std::chrono::milliseconds(rng.next_byte() % 2));
+               ++rounds;
 
-               // TODO send multiple parts
-               client.send(client_sent);
-               }
-
-            if(server.is_active() && server_sent.empty())
-               {
-               result.test_eq("server ALPN", server.next_protocol(), "test/3");
-
-               const size_t s_len = 1 + rng.next_byte() + rng.next_byte();
-               server_sent = unlock(rng.random_vec(s_len));
-               server.send(server_sent);
-               }
+               if(rounds > 100)
+                  {
+                  result.test_failure("Still here after many rounds");
+                  break;
+                  }
 
-            const bool corrupt_client_data = (r == 3 && rng.next_byte() % 3 <= 1 && rounds < 10);
-            const bool corrupt_server_data = (r == 4 && rng.next_byte() % 3 <= 1 && rounds < 10);
+               if(handshake_done && (client->is_closed() || server->is_closed()))
+                  break;
 
-            if(c2s_traffic.size() > 0)
-               {
-               /*
-               * Use this as a temp value to hold the queues as otherwise they
-               * might end up appending more in response to messages during the
-               * handshake.
-               */
-               std::vector<byte> input;
-               std::swap(c2s_traffic, input);
-
-               if(corrupt_server_data)
+               if(client->is_active() && client_sent.empty())
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if|for|do|while|switch statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=cpp%3AS134)